eng/scripts/Language-Settings.ps1

```diff
@@ -172,58 +172,97 @@ function Get-javascript-GithubIoDocIndex()
-    exit(1)
+# "@azure/package-name@1.2.3" -> "@azure/package-name"
+function Get-PackageNameFromDocsMsConfig($DocsConfigName) { 
+  if ($DocsConfigName -match '^(@?.+?)(@(.*))?$') { 
+    return $Matches[1]
   }
 
-  $allJson = Get-Content $pkgJsonLoc | ConvertFrom-Json
+  throw "Cannot find package $DocsConfigName"
+}
 
-  $visibleInCI = @{}
+function Update-javascript-DocsMsPackages($DocsRepoLocation)
+{
+  $publishedPackages = Get-CSVMetadata `
+  | Where-Object { $_.New -eq 'true' -and $_.Hide -ne 'true' }
+
+  # Update "Latest" packages
+  $latestConfigPath = Join-Path $DocRepoLocation '/ci-configs/packages-latest.json' -Resolve
+  Write-Host "Updating latest configuration: $latestConfigPath"
+  $latestPackages = Get-Content $latestConfigPath -Raw | ConvertFrom-Json
+
+  $gaPackages = $publishedPackages | Where-Object { $_.VersionGA.trim() }
+  foreach ($package in $gaPackages) { 
+    if ($latestPackages.npm_package_sources.name -contains $package.Package) { 
+      # Onboarded packages already contains this package, skip
+      Write-Host "Keeping onboarded package $($package.Package)"
+      continue
+    }
 
-  for ($i = 0; $i -lt $allJson.npm_package_sources.Length; $i++)
-  {
-    $pkgDef = $allJson.npm_package_sources[$i]
-    $accessor = ($pkgDef.name).Replace("`@next", "")
-    $visibleInCI[$accessor] = $i
+    Write-Host "Add GA package: $($package.Package)"
+    $latestPackages.npm_package_sources += @{ name = $package.Package }
   }
 
-  foreach ($releasingPkg in $pkgs)
-  {
-    $name = $releasingPkg.PackageId
-
-    if ($releasingPkg.IsPrerelease)
-    {
-      $name += "`@next"
+  $latestPackages | ConvertTo-Json -Depth 100 | Set-Content $latestConfigPath
+
+  # Update "preview" packages
+  $previewConfigPath = Join-Path $DocRepoLocation '/ci-configs/packages-preview.json' -Resolve
+  Write-Host "Updating preview configuration: $previewConfigPath"
+  $previewPackages = Get-Content $previewConfigPath -Raw | ConvertFrom-Json
+
+  $outputPackages = @()
+  foreach ($package in $previewPackages.npm_package_sources) {
+    # Do not filter by GA/Preview status because we want differentiate between
+    # tracked and non-tracked packages
+    $matchingPublishedPackages = $publishedPackages `
+    | Where-Object { $_.Package -eq (Get-PackageNameFromDocsMsConfig $package.name) }
+
+    # If this package does not match any published packages keep it in the list.
+    # This handles packages which are not tracked in metadata but still need to
+    # be built in Docs CI.
+    if (-not $matchingPublishedPackages) {
+      Write-Host "Keep non-tracked preview package: $($package.name)"
+      $outputPackages += $package
+      continue
     }
 
-    if ($visibleInCI.ContainsKey($releasingPkg.PackageId))
-    {
-      $packagesIndex = $visibleInCI[$releasingPkg.PackageId]
-      $existingPackageDef = $allJson.npm_package_sources[$packagesIndex]
-      $existingPackageDef.name = $name
+    $previewPackage = $matchingPublishedPackages `
```

Comment:
Lets assert that we don't have more then one here so that we don't accidently ignore a potential issue.